subsys/bootloader/Kconfig

```diff
@@ -6,9 +6,22 @@
 
 menu "Bootloader"
 
-config MCUBOOT_BUILD_S1_VARIANT
-	bool
-	prompt "Build S1 variant of mcuboot"
+config BUILD_S1_VARIANT
+	# This can not depend on BOOTLOADER_MCUBOOT, or SECURE_BOOT as this
+	# option has to be set when building MCUBoot (or any other child image
+	# which will be stored in S0 and S1) itself.
+	bool "Build S1 variant of specified image"
+	default y if SECURE_BOOT
+	help
+	  Build upgrade candidate of image for alternative slot S1.
+
+config S1_VARIANT_IMAGE_NAME
```

Comment:
Can this information be implied? From PM_ defines maybe, by seeing which image is in S0? How does it work if S0 contains both SPM and app?

Reply:
I imply it through these default values, IMO the variants are quite static, and will always be.

Remember that PM variables are not available until the end of cmake configure stage.

The case of SPM and app is not solved, and will not be in this PR, as there are other issues that needs to be fixed first.